external_file_location/models/attachment.py

```diff
@@ -28,6 +29,7 @@ def _run(self):
             location = self.location_id
             cls = protocols.get(location.protocol)[1]
             path = os.path.join(self.task_id.filepath, self.datas_fname)
-            with cls.connect(location) as conn:
-                datas = base64.decodestring(self.datas)
-                conn.setcontents(path, data=datas)
+            #with cls.connect(location) as conn:
+            conn = cls.connect(location)
+            datas = base64.decodestring(self.datas)
+            conn.setcontents(path, data=datas)
```

Comment:
If it's a file, it's [better to still close it](http://blog.lerner.co.il/dont-use-python-close-files-answer-depends/). I suggest adding something like:

```
try:
    conn.close()
except AttributeError:
    pass
```

at the end.